Source/WebCore/Modules/indexeddb/IDBTransaction.cpp

```diff
@@ -1253,25 +1253,35 @@ Ref<IDBRequest> IDBTransaction::requestPutOrAdd(IDBObjectStore& objectStore, Ref
     scheduleOperation(IDBClient::TransactionOperationImpl::create(*this, request.get(), [protectedThis = Ref { *this }, request] (const auto& result) {
         protectedThis->didPutOrAddOnServer(request.get(), result);
     }, [protectedThis = Ref { *this }, key, value = Ref { value }, overwriteMode, objectStoreInfo = objectStore.info()] (auto& operation) {
-        protectedThis->putOrAddOnServer(operation, objectStoreInfo, key.get(), value.ptr(), overwriteMode);
+        protectedThis->putOrAddOnServer(operation, objectStoreInfo, key.get(), value, overwriteMode);
     }), IsWriteOperation::Yes);
 
     return request;
 }
 
-void IDBTransaction::putOrAddOnServer(IDBClient::TransactionOperation& operation, const IDBObjectStoreInfo& objectStoreInfo, RefPtr<IDBKey> key, RefPtr<SerializedScriptValue> value, const IndexedDB::ObjectStoreOverwriteMode& overwriteMode)
+void IDBTransaction::putOrAddOnServer(IDBClient::TransactionOperation& operation, const IDBObjectStoreInfo& objectStoreInfo, const RefPtr<IDBKey>& key, const Ref<SerializedScriptValue>& value, const IndexedDB::ObjectStoreOverwriteMode& overwriteMode)
```

Comment:
We never pass Ref / RefPtr by const reference. It's either an r-value reference when transferring ownership or passing a raw `T&`.